### PR TITLE
Adding R Script and dataframe for greenspace

### PR DESCRIPTION
**Pull request by zaramorganoluwole**
- Added R Script to wrangle dataset for green space
- Added rda file on green space

### CODE REVIEW

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Load packages
+library(tidyverse)
+library(geographr)
+
+# Load lookup dataset
+lookup_ltla_raw <- lookup_ltla_ltla
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:6]: This variable isn't used in the rest of the code - if not using, should delete

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# Filter the dataset for Welsh local authorities - LTLA code and LTLA name
+raw_welsh_greenspace <- data.frame ()
+raw_welsh_greenspace <- lookup_ltla_raw |>
+  filter(startsWith(ltla21_code, "W0"))|>
+  select(ltla21_code, ltla21_name)
+print (raw_welsh_greenspace)
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:13]: Print statements are for you, as you are developing the script, to test or see what your code is doing. Code that isn't required in the processing of the dataset should be removed from the final script.

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Load packages
+library(tidyverse)
+library(geographr)
+
+# Load lookup dataset
+lookup_ltla_raw <- lookup_ltla_ltla
+
+# Filter the dataset for Welsh local authorities - LTLA code and LTLA name
+raw_welsh_greenspace <- data.frame ()
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:9]: `raw` is typically used as part of a variable name when you are loading a dataset raw, i.e. before it is tidied and clean. In this instance you are creating a new dataframe. See this --> http://adv-r.had.co.nz/Style.html#:~:text=Generally%2C%20variable%20names%20should%20be,this%20is%20not%20easy!).&text=Where%20possible%2C%20avoid%20using%20names,the%20readers%20of%20your%20code. on how to name objects, variables and functions. Note that this whole repo is about Wales so having `welsh` in the variable name may not be a good differentiator.

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+raw_welsh_greenspace <- lookup_ltla_raw |>
+  filter(startsWith(ltla21_code, "W0"))|>
+  select(ltla21_code, ltla21_name)
+print (raw_welsh_greenspace)
+
+# Add new columns for the green space data
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:15]: Could you please add a couple of lines of comment with the url where you got the data from (e.g. `#source: www...... `) and a comment per column explaining briefly what each column denotes. The full explanation will eventually be added to the dataframe's documentation, but we need a short one here.

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+raw_welsh_greenspace$PPP <- NA
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:16]: - Column names should make sense to the reader, similar to object names. For example now I do not know what the columns PPP or N10 denotes. Please rename. Please also use snake case.
- Are all columns required, for example the column `NUTS1`? What is it added to the dataset?

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+raw_welsh_greenspace$NUTS1 <- NA
+raw_welsh_greenspace$POP <- NA
+raw_welsh_greenspace$MIN <- NA
+
+raw_welsh_greenspace$PPP <- c(26.90, 72.81, 26.25, 100.19, 32.67, 47.17, 20.30, 45.06, 51.57, 35.32, 46.07, 18.68, 37.15, 23.61, 24.09, 46.45, 34.92, 24.84, 23.39, 20.24, 36.66, 23.40)
+raw_welsh_greenspace$N10 <- c(22108, 30094, 13826, 16744, 15135,14011, 31324, 34659, 50909, 25060, 25060, 17663, 7147, 8906, 13449, 15610, 3671, 5005, 22252, 24935, 49696, 3019)
+raw_welsh_greenspace$NUTS1 <- ("WALES")
+raw_welsh_greenspace$POP <- c(69931, 1258522, 119121, 95753, 157058, 134391, 70873, 126897, 189830, 248749, 143780, 148031, 136716, 370051, 242163, 18002, 68810, 94232, 96396, 160105, 133393, 60494)
+raw_welsh_greenspace$MIN <- c(0,1,0,1,0,1,0,1,1,1,1,0,1,0,0,1,1,0,0,0,1,0)
+# Save the df as an .rda in the data/ folder
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:27]: - Please use consistent spacing throughout your script. 
- Refer to this [guide ](https://r4ds.hadley.nz/workflow-style) on style (including spacing) and implement any recommended style changes.
- Install and use the package `styler` as recommended in the guide.

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+> save(raw_welsh_greenspace, file = "raw_welsh_greenspace.rda")
```
Comment (jennajt) [inline data-raw/healthy-places/access-to-green-space/access-greenspace.R:28]: - Please rename the datafile using the same rules as how you would name objects and functions.
- There are currently two `raw_welsh_greenspace.rda` files saved in the repo. There should only be one datafile saved. This file should be saved in the folder `health-index-wales/data/`. Consider that this data folder will eventually contain 40+ datasets across the three health domains. What is a good name for the dataset? For example should all datasets in this domain start with `healthy_places` or `hp` or `places`? 
- Please rewrite your code to save the dataset for conciseness. Please refer to any of the scripts in healthyr/data-raw for the function we have previously used to save datasets. We should use the same function for consistency.

**Review by jennajt — CHANGES_REQUESTED** [review]
Hi Zara, 

Well done on submitting your first script. You've done a good job and it a good first draft. Please see my comments on your script and make the required changes please.

Let me know when you are ready for it to be reviewed again. Once you make the changes on your laptop, you will need to add, commit and push these changes again. 

Any questions let me know.

File: data-raw/healthy-places/access-to-green-space/access-greenspace.R (hunk 3 above)
Comment (zaramorganoluwole) [reply data-raw/healthy-places/access-to-green-space/access-greenspace.R:9]: I changed it to "cleaned_greenspace"

**Pull request closed without merging**